lib/constants.go

```diff
@@ -101,7 +103,24 @@ const (
 	//
 	// At the time of this writing, the intent is to deploy it in a backwards-compatible
 	// fashion, with the eventual goal of phasing out blocks with the previous version.
-	HeaderVersion1       = uint32(1)
+	HeaderVersion1 = MsgDeSoHeaderVersion(1)
+	// This version introduces the transition from Proof of Work to Proof of Stake blocks.
+	// It includes several changes to the header format:
+	// - Nonce field is deprecated
+	// - ExtraNonce field is deprecated
+	// - ValidatorsVoteQC field is added
+	// - ValidatorsTimeoutAggregateQC field is added
+	//
+	// This format change is a breaking change that is not backwards-compatible with
+	// versions 0 and 1.
+	HeaderVersion2 = MsgDeSoHeaderVersion(2)
```

Comment:
I was also adding a new header version in one of my PRs. Will merge with your branch to make it consistent. One thing I considered is naming it something PoS-labeled, e.g. `HeaderPoSVersion1` or `PoSHeaderVersion1`. You know, adds respect :D

Reply:
Open to doing either. Slight preference to having the const name purely handle version incrementing and contextualize how to compare header versions. Ex: it's clear from the names that HeaderVersion2 comes after HeaderVersion0, whereas we lose that if we start from PoSHeaderVersion1 again.

If we need to identify PoS header versions explicitly, we can define a `MsgDeSoHeaderVersion.IsPoSHeader()` that returns true/false.

Reply:
Agree w/ @tholonious on this. header versioning shouldn't take into account our consensus mechanism, even in naming form.